modular_skyrat/modules/wall_fungus/code/wall_fungus_component.dm

```diff
@@ -0,0 +1,141 @@
+
+
+#define FUNGUS_STAGE_ONE 1
+#define FUNGUS_STAGE_TWO 2
+#define FUNGUS_STAGE_MAX 3
+
+/**
+ * A wall eating mushroom.
+ *
+ * This mushroom spreads to walls and eats em up! It can be removed with a welder. If left unchecked it will eat the whole wall.
+ */
+/datum/component/wall_fungus
+	/// How far has the fungus progressed on the affected wall? Percentage.
+	var/progression_percent = 0
+	/// How many percent do we increase each subsystem fire?
+	var/progression_step_amount = 1
+	/// What stage are we at?
+	var/progression_stage = FUNGUS_STAGE_ONE
+	/// Our overlay icon file
+	var/overlay_icon_file = 'modular_skyrat/modules/wall_fungus/icons/wall_fungus_overlay.dmi'
+	/// How likely are we to spread to another wall?
+	var/spread_chance = 1
+	/// How far can we spread?
+	var/spread_distance = 3 // Tiles
+	/// A reference to our parent wall.
+	var/turf/closed/wall/parent_wall
+	/// How likely are we to drop a shroom upon destruction?
+	var/drop_chance = 30
+
+/datum/component/wall_fungus/Initialize(override_progression_step_amount, override_spread_chance, override_spread_distance, override_drop_chance)
+	// This stuff enables badminery.
+	if(override_progression_step_amount)
+		progression_step_amount = override_progression_step_amount
+	if(override_spread_chance)
+		spread_chance = override_progression_step_amount
+	if(override_spread_distance)
+		spread_distance = override_spread_distance
+	if(override_drop_chance)
+		drop_chance = override_drop_chance
+
+	if(!iswallturf(parent))
+		return COMPONENT_INCOMPATIBLE
```

Comment:
You might want to move it to the start of the Initialization, that way it doesn't do anything unnecessary before returning.

Reply:
True